core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala

```diff
@@ -1053,7 +1053,8 @@ class KRaftClusterTest {
     val cluster = new KafkaClusterTestKit.Builder(
       new TestKitNodes.Builder().
         setNumBrokerNodes(1).
-        setNumControllerNodes(1).build()).build()
+        setNumControllerNodes(1).
+        setFeature(KRaftVersion.FEATURE_NAME, 1.toShort).build()).build()
```

Comment:
Yeah. This doesn't emulate or mock kraft.version correctly. Remember that kraft version are persisted and replicated using the KRaftVersion control record and not the FeatureLevelRecord metadata record.

This is why I didn't re-implement this in the cherry-pick but I forgot to run this suite.